### PR TITLE
Tagging 11 math

### PR DESCRIPTION
**Pull request by u-fischer**
This corrects two small bug (tagging-11 and tagging -05) in the latex-lab math code.

## Status of pull request

- Ready to merge

## Checklist of required changes before merge will be approved
- [x ] Test file(s) added
- [x ] Version and date string updated in changed source files
- [ ] Relevant `\changes` entries in source included (not relevant in latex-lab)
- [x ] Relevant `changes.txt` updated
- [ ] `ltnewsX.tex` (and/or `latexchanges.tex`) updated (not relevant)

### CODE REVIEW

File: required/latex-lab/changes.txt [hunk 1]
```diff
@@ -1,10 +1,15 @@
 2023-09-04 Ulrike Fischer <Ulrike.Fischer@latex-project.org>
-    * label-lab-new-or-1.dtx latex-lab-new-or-2.dtx
+	* latex-lab-math.dtx: fix below display skip, tagging issue #11
+	* latex-lab-math.dtx: allow to use it without tagging.
+	* latex-lab-math.dtx: avoid tagging as math if fakemath is detected, tagging issue #5.
```
Comment (car222222) [inline required/latex-lab/changes.txt:4]: Maybe add something like "temporary fix" ?

File: required/latex-lab/latex-lab-math.dtx [hunk 2]
```diff
@@ -310,11 +310,25 @@
       {
         \@@_process:nn { math } {#1} % $
 % fairly simple this one
```
Comment (car222222) [inline required/latex-lab/latex-lab-math.dtx:312]: Delete this line.

File: required/latex-lab/latex-lab-math.dtx [hunk 3]
```diff
@@ -310,11 +310,25 @@
...
-        \tagmcend %end P-chunk, in code: \tag_mc_end_push:
-        \@kernel@math@begin
-        #1 $
-        \@kernel@math@end
-        \tagmcbegin{}  % restart P-chunk (whatsits in pdftex)
+%    \end{macrocode}
+% We do not want math tagging in fakemath or when measuring, 
+% so we imitate the test inside
+% \cs{@@_process:nn} for now, see https://github.com/latex3/tagging-project/issues/5
```
Comment (car222222) [inline required/latex-lab/latex-lab-math.dtx:316]: As it says, OK for now!  

Further work is needed here, particularly for some edge cases, but, for now:  Looks good to me.

File: required/latex-lab/latex-lab-math.dtx [hunk 4]
```diff
@@ -335,16 +349,22 @@
         $$
       }
   }
-
+%    \end{macrocode}
+% To allow to use the code without tagging we guard. But probably tagpdf should
```
Comment (car222222) [inline required/latex-lab/latex-lab-math.dtx:353]: Looks good; but I have not checked whether this is the only place in the code where such a guard is needed.

File: required/latex-lab/latex-lab-math.dtx [hunk 5]
```diff
@@ -814,12 +837,23 @@
...
   }
+
 \exp_args:No \tex_everydisplay:D
   {
     \tex_the:D \tex_everydisplay:D
     \iftrue  % this may have to be a settable flag!
 %      {
 %        \typeout{==>~ in~ everydisplay}
+%    \end{macrocode}
+% flipping the \cs{belowdisplay} values is done so that we get (assumption)
```
Comment (car222222) [inline required/latex-lab/latex-lab-math.dtx:848]: All the messing with these skips seems totally weird to me!  

Is there a better way? 

These changes clearly improve the situation.

**Review by car222222 — APPROVED** [review]
No real problems found.  But what do I know?

Maybe document that fixes are probably still needed, at least for some (very) edge cases?

**Review by josephwright — APPROVED** [review]

File: required/latex-lab/latex-lab-math.dtx (hunk 2 above)
Comment (muzimuzhi) [reply required/latex-lab/latex-lab-math.dtx:312]: Or, this comment line should be surrounded by `%    \end{macrocode}` and `%    \begin{macrocode}`?

**Comment by car222222** [comment]
Could do that, but it is wrong.

Why was nothing at all done about it? 

Also, do something about this:

"document that fixes are probably still needed".

**Comment by car222222** [comment]
Why was the branch deleted before completion of all these?

**Comment by car222222** [comment]
What is the point of asking for a review if it gets totally ignored?

File: required/latex-lab/latex-lab-math.dtx (hunk 2 above)
Comment (muzimuzhi) [reply required/latex-lab/latex-lab-math.dtx:312]: > Could do that, but it is wrong.
>
> Why was nothing at all done about it?

Sorry, I deleted my comment shortly after I realized that you're right. The original comment was

> Or, this comment line should be surrounded by `%    \end{macrocode}` and `%    \begin{macrocode}`?

**Pull request merged**